src/custom-element.js

```diff
@@ -301,6 +286,26 @@ function createBaseCustomElementClass(win) {
       /** @private {?./layout-delay-meter.LayoutDelayMeter} */
       this.layoutDelayMeter_ = null;
 
+      // Closure compiler appears to mark HTMLElement as @struct which
+      // disables bracket access. Force this with a type coercion.
+      const nonStructThis = /** @type {!Object} */ (this);
+
+      // `opt_implementationClass` is only used for tests.
+      let Ctor =
+        win.__AMP_EXTENDED_ELEMENTS &&
+        win.__AMP_EXTENDED_ELEMENTS[this.elementName()];
+      if (getMode().test && nonStructThis['implementationClassForTesting']) {
+        Ctor = nonStructThis['implementationClassForTesting'];
+      }
+      devAssert(Ctor);
+      try {
+        /** @private {!./base-element.BaseElement} */
+        this.implementation_ = new Ctor(this);
```

Comment:
I'm also happy to do this in the CE's `connectedCallback` after getting the ampdoc. There should be little downside.

Reply:
Let's do that, but in a separate PR.